### ISSUE TITLE
Incompatible with Python >= 3.11

### ISSUE THREAD

**Issue by dlbas** [aliev/aioauth#68]
Hi @aliev, thanks for the package! It's just what I wanted for one of my personal projects.
Found an unfortunate bug due to using Python 3.11.

I believe this is due to python's `dataclasses` module changed the way of checking if the object is mutable.

In [3.11](https://github.com/python/cpython/blob/3.11/Lib/dataclasses.py#L814)
```python
    if f._field_type is _FIELD and f.default.__class__.__hash__ is None:
        raise ValueError(f'mutable default {type(f.default)} for field '
                         f'{f.name} is not allowed: use default_factory')
```

In [3.10](https://github.com/python/cpython/blob/3.10/Lib/dataclasses.py#L811)
```python
    if f._field_type is _FIELD and isinstance(f.default, (list, dict, set)):
        raise ValueError(f'mutable default {type(f.default)} for field '
                         f'{f.name} is not allowed: use default_factory')

```

It should be a pretty straightforward fix and I'll be more than happy to help

## Expected Result
Be able to import the package.

## Actual Result

Import failed with the exception (in the next paragraph).

## Reproduction Steps

```
> make test

pytest tests
ImportError while loading conftest '/Users/daniel/opensource/aioauth/tests/conftest.py'.
tests/conftest.py:7: in <module>
    from aioauth.grant_type import (
aioauth/grant_type.py:11: in <module>
    from .errors import (
aioauth/errors.py:18: in <module>
    from .requests import TRequest
aioauth/requests.py:62: in <module>
    @dataclass
../../.pyenv/versions/3.11.1/lib/python3.11/dataclasses.py:1220: in dataclass
    return wrap(cls)
../../.pyenv/versions/3.11.1/lib/python3.11/dataclasses.py:1210: in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
../../.pyenv/versions/3.11.1/lib/python3.11/dataclasses.py:958: in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
../../.pyenv/versions/3.11.1/lib/python3.11/dataclasses.py:815: in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
E   ValueError: mutable default <class 'aioauth.collections.HTTPHeaderDict'> for field headers is not allowed: use default_factory
make: *** [test] Error 4
```

## System Information

```
> python --version
Python 3.11.1

> pip show aioauth
Name: aioauth
Version: 1.5.0
Summary: Asynchronous OAuth 2.0 framework for Python 3.
Home-page: https://github.com/aliev/aioauth
Author: Ali Aliyev
Author-email: ali@aliev.me
License: The MIT License (MIT)
Location: /Users/daniel/opensource/aioauth
Requires: typing_extensions
Required-by: 
```

**Comment by aliev** [comment]
Hi, Daniel

Thanks for good catch! I will solve this issue asap.

**Comment by aliev** [comment]
@dlbas Version v1.5.1 had been released with Python 3.11 support. Should work now

**Comment by dlbas** [comment]
Thanks! That was a quick one 👍
